src/components/IDO/IDO.tsx

```diff
@@ -0,0 +1,282 @@
+                    return name === token.symbol
+                  })
+                )
+              }}
+              onMaxClick={() => {
+                if (tokenFromIndex !== null) {
+                  setAmountFrom(
+                    printBN(tokens[tokenFromIndex].balance, tokens[tokenFromIndex].decimal)
+                  )
+                }
+              }}
+            />
+            <Grid>
+              <Typography className={classes.tokenComponentText}>Deposited amount:</Typography>
+            </Grid>
+            <Box className={classes.valueBox}>
+              <CardMedia image={invariantLogo} className={classes.iconLogo} />
+              <Box className={classes.infoContainer}>
+                <Box>
+                  <Typography className={classes.depositAmouted}>
+                    {totalDeposited.toFixed(2)} xUSD
+                  </Typography>
+                </Box>
+                <Box className={classes.boxInfo}>
+                  <Typography className={classes.valueInfo}>
+                    {currencyData.tether?.usd > 0
+                      ? `${(totalDeposited / currencyData.tether?.usd).toFixed(2)} USD`
+                      : '0.0 USD'}
+                  </Typography>
+                  <Typography className={classes.valueInfo}>
+                    {currencyData.solana?.usd > 0
+                      ? `${(totalDeposited / currencyData.solana?.usd).toFixed(2)} SOL`
+                      : '0.0 SOL'}
+                  </Typography>
+                  <Typography className={classes.valueInfo}>
+                    {currencyData.ethereum?.usd > 0
+                      ? `${(totalDeposited / currencyData.ethereum?.usd).toFixed(4)} xETH`
+                      : '0.0 xETH'}
+                  </Typography>
+                  <Typography className={classes.valueInfo}>
+                    {currencyData.bitcoin?.usd > 0
+                      ? `${(totalDeposited / currencyData.bitcoin?.usd).toFixed(4)} xBTC`
+                      : '0.0 xBTC'}
+                  </Typography>
+                </Box>
+              </Box>
+            </Box>
+            <OutlinedButton
+              name={getButtonMessage()}
+              className={classes.buttonBuy}
+              disabled={
+                getButtonMessage() === 'Insufficient trade volume' ||
+                getButtonMessage() === 'Insufficient balance'
+              }
+              color='primary'
+              onClick={() => {}}
+            />
+          </Grid>
+        </Box>
+      </Grid>
+      <Grid item sm={4} className={classes.info}>
+        <Box className={classes.containerInfo}>
+          <Typography className={classes.labelInfo}>Sale period ends in</Typography>
+          <Box className={classes.wrapperInfo}>
+            <CardMedia image={watchIcon} className={classes.icon} />
+            <Typography className={classes.textInfo}>
+              <AnimatedNumber value={15} formatValue={formatValue} />:
```

Comment:
time values should be also set from props, you could for example use setInterval in story to decrease value by one second

Reply:
I pass data from props, could you show how I can decrease timer in story, because my Component does not react to changes in props data, when I use setInterval functions in story. Can you help me with this problem?

Reply:
Tip: try using storybook hooks. It might be useful to read some articles about using setInterval in React, it can be kinda tricky. Also, look what will happen, if number you're passing through props has only one digit. Write a comment if you're still be having any problems.

Reply:
Thanks for the tip, I create a hook in the storybook and it works great :)

Reply:
I thought about writing solution for this in a slightly different way, but your code also looks neat and gets the job done,